server/routerlicious/packages/services-utils/src/scheduleJob.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+import * as cron from "node-cron";
+import { ILogger } from "@fluidframework/server-services-core";
+import { Provider } from "nconf";
+
+export async function scheduleJob<T>(
+    api: () => Promise<T>,
+    config: Provider,
+    featureGateConfigName: string,
+    callName: string,
+    cronExpressionConfigName: string,
+    logger?: ILogger,
+): Promise<any> {
+    let task;
+    if (config.get(featureGateConfigName)) {
+        const cronExpression = config.get(cronExpressionConfigName);
+        try {
+            if (cron.validate(cronExpression)) {
+                task = cron.schedule(cronExpression, api);
+                logger?.info(`Task ${callName} is scheduled, cron expression is ${cronExpression}`);
+                Lumberjack.info(`Task ${callName} is scheduled, cron expression is ${cronExpression}`);
+            } else {
+                return Promise.reject(new Error(`Invalid cron expression ${cronExpression}`));
```

Comment:
Using `return Promise.reject` here instead of `throw` will cause this to not be caught and logged in the catch block below. Is that intentional?

Reply:
Oh it was unintentional, throwing it might make it neater, will fix